### PR TITLE
Set input_pos after init

### PR DESCRIPTION
**Pull request by riewert**
This should fix the homing issues described in #547 .

### CODE REVIEW

File: Firmware/MotorControl/axis.cpp [hunk 1]
```diff
@@ -397,29 +397,29 @@ bool Axis::run_homing() {
...
 
-    // pos_setpoint is the starting position for the trap_traj so we need to set it.
-    controller_.pos_setpoint_ = min_endstop_.config_.offset;
-    controller_.vel_setpoint_ = 0.0f;  // Change directions without decelerating
-
-    // Set our current position in encoder counts to make control more logical
-    encoder_.set_linear_count((int32_t)(controller_.pos_setpoint_ * encoder_.config_.cpr));
-
+    // Calculate the desired position after offset.
+    float inputBuffer = encoder_.pos_estimate_.any().value() + min_endstop_.config_.offset;
```
Comment (samuelsadok) [inline Firmware/MotorControl/axis.cpp]: We need to check if `encoder_.pos_estimate_.any()` is `std::nullopt`. If so, `.value()` shouldn't be called and we need to abort homing.
`inputBuffer` => `input_buffer`

**Review by samuelsadok — CHANGES_REQUESTED** [review]
Thanks for the fix! Just a minor comment from my side.

File: Firmware/MotorControl/axis.cpp (hunk 1 above)
Comment (riewert) [reply Firmware/MotorControl/axis.cpp]: Added an error for the abort case, and fixed my naming convention. Too much Java lately...

See latest [commit](https://github.com/odriverobotics/ODrive/pull/550/commits/d68dfbda824ffb336dccf5eb3ee2c602998c7551).

File: Firmware/MotorControl/axis.cpp [hunk 2]
```diff
@@ -397,29 +397,34 @@ bool Axis::run_homing() {
...
     error_ &= ~ERROR_MIN_ENDSTOP_PRESSED; // clear this error since we deliberately drove into the endstop
 
-    // pos_setpoint is the starting position for the trap_traj so we need to set it.
-    controller_.pos_setpoint_ = min_endstop_.config_.offset;
-    controller_.vel_setpoint_ = 0.0f;  // Change directions without decelerating
-
-    // Set our current position in encoder counts to make control more logical
-    encoder_.set_linear_count((int32_t)(controller_.pos_setpoint_ * encoder_.config_.cpr));
+    
+    if (encoder_.pos_estimate_.any() == std::nullopt || !encoder_.pos_estimate_.any().has_value()){
```
Comment (samuelsadok) [inline Firmware/MotorControl/axis.cpp]: There's a possible race condition here: `encoder_.pos_estimate_.any()` could become `std::nullopt` between this line and the next. To fix this you can read `encoder_.pos_estimate_.any()` into a local variable and then operate on that.

Comment (riewert) [reply Firmware/MotorControl/axis.cpp]: Thanks for pointing that out. Modified it [here](https://github.com/odriverobotics/ODrive/pull/550/commits/7ac7c6ec58c6eabe7d832c7c9b105504c66c4242).

Comment (samuelsadok) [reply Firmware/MotorControl/axis.cpp]: Great, thanks! Looks good now.

**Review by samuelsadok — APPROVED** [review]

**Pull request merged**